pc_hal/conioHistory.hpp

```diff
@@ -0,0 +1,152 @@
+#ifndef __CONIOHISTORY_HPP__
+#define __CONIOHISTORY_HPP__
+
+#include <stdint.h>
+#include <string>
+#include <vector>
+
+/**
+ * @brief Starting file size for the history's log file
+ *
+ */
+#define INITIAL_FILE_SIZE 4096
+/**
+ * @brief Resize the file by a factor of FILE_RESIZE_FACTOR
+ * 
+ */
+#define FILE_RESIZE_FACTOR 2
+/**
+ * @brief Upon reaching a size of 32MB, increase file size in steps of 32MB
+ * 
+ */
+#define LINEAR_STEP_RESIZE 32 * 1024 * 1024
+/**
+ * @brief Max size of a single log file is 1GB
+ * 
+ */
+#define MAX_FILE_SIZE 1024 * 1024 * 1024
+
+/**
+ * @brief Struct that contains the necessary metadata for a line in the history
+ * 
+ */
+typedef struct CONIO_history_line_
+{
+    /**
+     * @brief Starting offset within the file
+     * 
+     */
+    size_t offset;
+    /**
+     * @brief Length of the line
+     * 
+     */
+    size_t len;
+    /**
+     * @brief Display flag: 0 do not display, 1 display to window
+     * 
+     */
+    uint32_t display : 1;
+    /**
+     * @brief More Framgents flag: 0 no successive fragments, 1 more fragments for the line
+     * 
+     */
+    uint32_t more_frag : 1;
+    /**
+     * @brief Fragment sequence number within the line
+     * 
+     */
+    uint32_t frag_seq : 30;
+    // Additional metadata may be necessary
+
+    /**
+     * @brief Construct a new conio history line object with manually set parameters
+     * 
+     * @param o Starting offset
+     * @param l Current length of the line
+     * @param d Display flag
+     * @param mf More Fragments flag
+     * @param fs Fragment sequence number
+     */
+    CONIO_history_line_(size_t o, size_t l, uint32_t d, uint32_t mf, uint32_t fs) : offset(o), len(l), display(d), more_frag(mf), frag_seq(fs) {}
+    /**
+     * @brief Construct a new conio history line object with default flags
+     * 
+     * @param o Starting offset
+     */
+    CONIO_history_line_(size_t o) : offset(o), len(0), display(0), more_frag(0), frag_seq(0) {}
+} CONIO_hist_line;
+
+namespace conioHistory
```

Comment:
Convert to global singleton to be consistent with rest of project.  Class can still be inside a namespace